usecases/scheduled_execution/offloading_job.go

```diff
@@ -261,10 +230,17 @@ func (w OffloadingWorker) Work(ctx context.Context, job *river.Job[models.Offloa
 		// If we did not get a multiple of save point batch size, we still have len(rules)%batch_size
 		// items to persist.
 		if idx%w.config.SavepointEvery > 0 {
-			remainingItems := offloadedIds[:idx%w.config.SavepointEvery]
+			span.AddEvent("partial_savepoint_reached")
+
+			remainingItemIds := offloadedIds[:idx%w.config.SavepointEvery]
+			remainingRules := rulesToOffload[:idx%w.config.SavepointEvery]
+
+			if err := w.writeBatchToBlobStorage(ctx, req, remainingRules, remainingItemIds); err != nil {
+				return stopChannelAndReturn(err)
```

Comment:
```suggestion
				return err
```
Just this should be good here (my "fix" on channeling errors back to the rule streaming goroutine I think only needs to be used for errors within the loop on rules

Reply:
(though I don't think this panics or anything)